src/merge.py

```diff
@@ -16,7 +16,19 @@ def merge(x: list[int], y: list[int]) -> list[int]:
     # FIXME: fill out the loop so you merge the lists
     # until one of them is empty
     while i < len(x) and j < len(y):
-        break  # FIXME: you shouldn't just break here
+        if x[i] < y[j]:
+            z.append(x[i])
+            i += 1
+        else:
+            z.append(y[j])
+            j +=1
+        # FIXME: you shouldn't just break here
     # At least one of the lists is empty now. Copy the
     # remainder of the other into z.
+    if i == len(x):
+        for n in y[j:]:
+            z.append(n)
+    else: 
+        for n in x[i:]:
+            z.append(n)
```

Comment:
The method `z.extend(y[j:])` does the same as

```python
for n in y[j:]:
    z.append(n)
```

so using it might be a little more readable.

Also, you do not need to check which list is fully processed. If you extend with `x[i:]` and `y[j:]`, then at least one of them is empty, and extending with an empty list doesn't do anything. You can safely extend both lists.